### DISCUSSION TITLE
Migration from Task Manager to Process Hacker

### DISCUSSION THREAD

**Discussion by amymor** [winsiderss/systeminformer#1356] [Questions / Answers] [answered]
hi, i wanna migrate from task manager to process hacker, but i cant understand something, i have some questions:

1. can PH have separate tabs for Processes and Details like task manger? its so useful because in Processes tab you can see overall info like CPU usage and memory usage and GPU usage and if you need more details like PID and elevated state and etc simply switch to Details tab. i didnt find any Details tab in PH, guess i should add all details column to main Processes tab and scroll horizontally, but this is what i hate, because horizontal scrollbar takes one row and its bad in my wide monitor. i just found switching between tabs is easier.

2. is there any way to show processes in 3 groups like task manger (Apps - Background processes  - Windows processes )?

3. how i can determine CPU usage for each app? PH shows about 80-90 for system idle process, what is it stand for? also i found process grouping is weird, i mean if i collapse a group it only shows me cpu usage of header instead sum of all processes in that group.

4. i cant find any columns that show me memory usage like task manager. i just see Private bytes column that shows 7.55 GB. is there any way to show memory usage in present like task manager?
thanks

**Comment by dmex** [comment, marked as answer]
> separate tabs for Processes and Details like task manger?

The processes tab in Task Manager only shows visible windows on the desktop and groups a few services. We're showing everything by default while TM is filtering everything out on the processes tab.

> is there any way to show processes in 3 groups like task manger

Processes are grouped based on the child processes.

> PH shows about 80-90 for system idle process, what is it stand for?

https://en.wikipedia.org/wiki/System_Idle_Process

> if i collapse a group it only shows me cpu usage of header instead sum of all processes in that group.

Including usage of children is disabled by default. You need to enable the `Include usage of collapsed processes` option in the settings:
![image](https://user-images.githubusercontent.com/1306177/188510509-d424b05e-930d-4345-8e08-6179db645c5c.png)

> i cant find any columns that show me memory usage like task manager. i just see Private bytes column that shows 7.55 GB. 

Add the `Private WS` column to see the same memory shown by Task Manager. Any columns ending with `Commit` or `WS` are generally what you'll see shown by TM.

> should add all details column to main Processes tab and scroll horizontally, but this is what i hate, because horizontal scrollbar takes one row and its bad in my wide monitor

Use Task Manager when you want limited/overview information and use SI/PH when you need more detailed information. You can launch TM from our main menu > Tools > Start Task Manager option.

You'll have to keep using Task Manager anyway because Microsoft blocks third party software (i.e. this project) from using functions and features implemented into Task Manager such as:

* Always on top: Microsoft added a new capability to always-on-top using window bands but Microsoft restricted the feature so it can only be called by Task Manager.
* App history: Windows records statistics for every program 24/7 but Microsoft restricted the feature so it can only be shown by Task Manager.
* Startup entries: Microsoft added a new startup API for store applications but Microsoft restricted the feature so only Task Manager can show those startup entries (even Microsoft's Sysinternals Autoruns utility can't show them).
* Power usage/Enterprise context/other functions are also restricted...

**Reply by DrewNeilson** [reply]
> You'll have to keep using Task Manager anyway because Microsoft blocks third party software (i.e. this project) from using functions and features implemented into Task Manager such as:
> 
> * Always on top: Microsoft added a new capability to always-on-top using window bands but Microsoft restricted the feature so it can only be called by Task Manager.
> * App history: Windows records statistics for every program 24/7 but Microsoft restricted the feature so it can only be shown by Task Manager.
> * Startup entries: Microsoft added a new startup API for store applications but Microsoft restricted the feature so only Task Manager can show those startup entries (even Microsoft's Sysinternals Autoruns utility can't show them).
> * Power usage/Enterprise context/other functions are also restricted...

Can you cite some sources that validate your statements about functions that Microsoft has restricted to Task Manager? I'd like to track over time the resource usage of individual applications as a percentage of total available system resources. For example, the percent of CPU time used by Microsoft Edge over the past 30 days.

**Comment by amymor** [comment]
thank you for great information.❤️
shame on Microsoft, i was trying to use classic theme that breaks TM ( another reason to hate Microsoft, lol), so i thought about PH because i found it powerful app, but as i understand the purpose is little different. also im used to using TM so cant like another GUI so as you said i stick to combination of TM and PH.
